java/arcs/core/data/recipe.proto

```diff
@@ -68,19 +68,21 @@ message HandleConnectionSpecProto {
   }
   // Identifies a connection in a particle spec.
   string name = 1;
-  Direction directions = 2;
+  Direction direction = 2;
   TypeProto type = 3;
 }
 
 // Representation of a type in Arcs.
 message TypeProto {
   oneof data {
     PrimitiveTypeProto primitive = 1;
-    EntityTypeProto entity = 2;
-    CollectionTypeProto collection = 3;
-    ReferenceTypeProto reference = 4;
```

Comment:
I am not sure how important this is, but don't you lose the ability to attach additional information with each type if needed later? This change seems like it would make it hard to modify the protos later in a backward compatible way. 

Also, should we working on the protos other than deserializing it into native structures?

Reply:
Yes, that's the tradeoff.

Staying backwards compatible is not a requirement for now - we don't persist these protos anywhere for long enough time to need to worry about it. Still, making breaking changes will require a bit of coordination - we can talk offline.

Should we be working on the protos? Likely not, but it would be nice if we could write out a proto in a human readable way (see #4768) for many reasons.

I may be optimizing for human readability here, which may be the wrong tradeoff. What's your gut instinct?